vpd-tool/include/vpd_tool.hpp

```diff
@@ -0,0 +1,51 @@
+#pragma once
+
+#include <string>
+
+namespace vpd
+{
+/**
+ * @brief Class to support operations on VPD.
+ *
+ * The class provides API's to,
+ * Read keyword value from DBus/hardware.
+ * Update keyword value to DBus/hardware.
+ * Dump DBus object's critical information.
+ * Fix system VPD if any mismatch between DBus and hardware data.
+ * Reset specific system VPD keywords to its default value.
+ * Force VPD collection for hardware.
+ */
+class VpdTool
+{
+  public:
+    /**
+     * @brief Read keyword value.
+     *
+     * API to read VPD keyword's value from the given input path.
+     * If the provided path is EEPROM path, read keyword's value from the
+     * hardware. Othrwise if the provided path is DBus path, read keyword's
+     * value from DBus.
+     *
+     * @param[in] i_fruPath - DBus object path or EEPROM path.
+     * @param[in] i_recordName - Record name.
+     * @param[in] i_keywordName - Keyword name.
+     * @param[in] i_onHardware - True if i_fruPath is EEPROM path, false
+     * otherwise.
+     * @param[in] i_fileToSave - To hold whether result should redirect to a
+     * file or console.
+     */
+    void readKeyword(const std::string& i_fruPath,
+                     const std::string& i_recordName,
+                     const std::string& i_keywordName, bool i_onHardware,
+                     const std::string& i_fileToSave = {});
+
+  private:
+    const std::string m_vpdManagerService = "com.ibm.VPD.Manager";
```

Comment:
a) Either have a constant file
or,
b) Use constexpr in cpp file if it is local only.

Reply:
moved under new file constants.hpp .